### PR TITLE
[SYCL] Update tests for default context extension

### PR DESCRIPTION
**Pull request by alexbatashev**
Some tests assume that two default-created queues will be bound to two different contexts. This is no longer true with default contexts extension. Update tests to explicitly create distinct contexts.

### CODE REVIEW

File: SYCL/Basic/buffer/buffer_dev_to_dev.cpp [hunk 1]
```diff
@@ -25,9 +25,13 @@ int main() {
                           {property::buffer::use_host_ptr()});
 
     default_selector Selector;
+    device Device(Selector);
 
-    queue FirstQueue(Selector);
-    queue SecondQueue(Selector);
+    context FirstContext(Device);
+    context SecondContext(Device);
```
Comment (s-kanaev) [inline SYCL/Basic/buffer/buffer_dev_to_dev.cpp:31]: Minor NIT question: is it possible to use default ctx instead of one of explicit ones?

**Review by s-kanaev — APPROVED** [review]
LGTM

File: SYCL/Basic/buffer/buffer_dev_to_dev.cpp (hunk 1 above)
Comment (alexbatashev) [reply SYCL/Basic/buffer/buffer_dev_to_dev.cpp:31]: Yes, if we have two distinct devices (which is not always the case). The reason test requires two different contexts is that it forces us to move data from device to device through host.

**Review by romanovvlad — APPROVED** [review]

**Comment by alexbatashev** [comment]
@smaslov-intel friendly ping

**Review by smaslov-intel — APPROVED** [review]

**Pull request merged**